mirage/utils/siaf_interface.py

```diff
@@ -0,0 +1,145 @@
+"""SIAF interface module to support accessing SIAF information.
+
+This module provides ``mirage`` with functions to interface SIAF content via the pysiaf module.
+
+Authors
+-------
+
+    - Johannes Sahlmann
+
+Use
+---
+
+    This module can be imported and used with
+
+    ::
+
+        from mirage.utils import siaf_interface
+
+"""
+
+import numpy as np
+
+import pysiaf
+from pysiaf import iando
+from ..utils import rotations
+from ..utils import set_telescope_pointing_separated as set_telescope_pointing
+
+
+def get_siaf_information(instrument, aperture, ra, dec, telescope_roll):
+    """Use pysiaf to get aperture information
+
+    Parameters
+    ----------
+    instrument : str
+        Instrument name.
+
+    aperture : str
+        Aperture name (e.g. "NRCA1_FULL")
+    """
+    siaf = pysiaf.Siaf(instrument)[aperture]
+    local_roll = set_telescope_pointing.compute_local_roll(telescope_roll,
+                                                           ra, dec, siaf.V2Ref,
+                                                           siaf.V3Ref)
+    # Create attitude_matrix
+    att_matrix = rotations.attitude(siaf.V2Ref, siaf.V3Ref, ra, dec, local_roll)
+
+    # Get full frame size
+    fullframesize = siaf.XDetSize
+
+    # Subarray boundaries in full frame coordinates
+    xcorner, ycorner = sci_subarray_corners(instrument, aperture)
+    subarray_boundaries = [xcorner[0], ycorner[0], xcorner[1], ycorner[1]]
+    return siaf, local_roll, att_matrix, fullframesize, subarray_boundaries
+
+
+def sci_subarray_corners(instrument, aperture_name, verbose=False):
+    """Return the two opposing aperture corners in the SIAF Science frame of the full-frame SCA.
+
+    This function serves as interface between the SIAF information accessible via the pysiaf package
+    and the subarray information formatted for use by mirage.
+
+    Parameters
+    ----------
+    instrument : str
+        JWST instrument name with correct capitalization
+    aperture_name : str
+        SIAF aperture name
+    verbose : bool
+        Verbose output on/off
+
+    Returns
+    -------
+    x_sci, y_sci : tuple of numpy arrays
+        Subarray corner coordinates
+
+    """
+    # get SIAF
+    siaf = pysiaf.Siaf(instrument)
+
+    # get master aperture names
+    siaf_detector_layout = iando.read.read_siaf_detector_layout()
+    master_aperture_names = siaf_detector_layout['AperName'].data
+
+    # read pysiaf aperture definition file containing DetRef and SciRef values
+    siaf_aperture_definitions = iando.read.read_siaf_aperture_definitions(instrument)
```

Comment:
This line returns the error: 
```
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lchambers/miniconda3/envs/OTECommSims/lib/python3.5/site-packages/pysiaf/source_data/FGS/fgs_siaf_aperture_definition.txt'
```
But I'm not sure how the `pysiaf.iando` module works, so I'm not sure how to fix it. Is this supposed to be pointing to a aperture definition file under `$MIRAGE_DATA`, rather than under my local package installation?

Reply:
Try updating your pysiaf installation. Johannes just released a new version with the FGS file last week.